### PR TITLE
neoss: init at 1.1.8

### PR DESCRIPTION
**Pull request by ghost**
## Description of changes

Add new package `neoss`

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/by-name/ne/neoss/package.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    hash = "sha256-1palorAZ58eHWAyQwfrz1RJGm8kwmR3UwvfI4bqstN0=";
+  };
+
+  npmDepsHash = "sha256-jmFhkV80D8SQcC3pRWg2FfQ0v3ur/gChAELpa66KpL8=";
+
+  meta = with lib; {
+    description = "User-friendly and detailed socket statistics with a Terminal UI";
+    homepage = "https://github.com/PabloLec/neoss";
+    license = licenses.bsd3;
+    maintainers = with maintainers; [ snowflake ];
```
Comment (eclairevoyant) [inline pkgs/by-name/ne/neoss/package.nix:23]: Please set `meta.mainProgram`

**Review by eclairevoyant — CHANGES_REQUESTED** [review]
Also how is this meant to be configured? I get the following when running:
```
$ result/bin/neoss
id: ‘100’: no such user
id: ‘100’: no such user
Error: Command failed: id -nu 100
id: ‘100’: no such user

    at checkExecSyncError (node:child_process:890:11)
    at execSync (node:child_process:962:15)
    at /nix/store/8vzvnv0zwn3qc3h25inq313hh5sapm6v-neoss-1.1.8/lib/node_modules/neoss/build/utils/users.js:83:50
    at Generator.next (<anonymous>)
    at /nix/store/8vzvnv0zwn3qc3h25inq313hh5sapm6v-neoss-1.1.8/lib/node_modules/neoss/build/utils/users.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/nix/store/8vzvnv0zwn3qc3h25inq313hh5sapm6v-neoss-1.1.8/lib/node_modules/neoss/build/utils/users.js:4:12)
    at getUserData (/nix/store/8vzvnv0zwn3qc3h25inq313hh5sapm6v-neoss-1.1.8/lib/node_modules/neoss/build/utils/users.js:72:12)
    at /nix/store/8vzvnv0zwn3qc3h25inq313hh5sapm6v-neoss-1.1.8/lib/node_modules/neoss/build/utils/getStats.js:53:58
```

**Comment by snf1k** [comment]
> Also how is this meant to be configured? I get the following when running:
> 
> ```
> $ result/bin/neoss
> id: ‘100’: no such user
> id: ‘100’: no such user
> Error: Command failed: id -nu 100
> id: ‘100’: no such user
> 
>     at checkExecSyncError (node:child_process:890:11)
>     at execSync (node:child_process:962:15)
>     at /nix/store/8vzvnv0zwn3qc3h25inq313hh5sapm6v-neoss-1.1.8/lib/node_modules/neoss/build/utils/users.js:83:50
>     at Generator.next (<anonymous>)
>     at /nix/store/8vzvnv0zwn3qc3h25inq313hh5sapm6v-neoss-1.1.8/lib/node_modules/neoss/build/utils/users.js:8:71
>     at new Promise (<anonymous>)
>     at __awaiter (/nix/store/8vzvnv0zwn3qc3h25inq313hh5sapm6v-neoss-1.1.8/lib/node_modules/neoss/build/utils/users.js:4:12)
>     at getUserData (/nix/store/8vzvnv0zwn3qc3h25inq313hh5sapm6v-neoss-1.1.8/lib/node_modules/neoss/build/utils/users.js:72:12)
>     at /nix/store/8vzvnv0zwn3qc3h25inq313hh5sapm6v-neoss-1.1.8/lib/node_modules/neoss/build/utils/getStats.js:53:58
> ```

Which operating system you're running on?

**Comment by eclairevoyant** [comment]
> Which operating system are you running on? This works fine under Linux

I am running this on NixOS, x86_64-linux.

File: pkgs/by-name/ne/neoss/package.nix (hunk 1 above)
Comment (snf1k) [reply pkgs/by-name/ne/neoss/package.nix:23]: OK, I added it.

**Comment by snf1k** [comment]
The software doesn't seem to work with NixOS, I only tested it with Arch Linux.
I don't know much JavaScript either, so I can't help it.
I found the software at first and thought it was good, so I wanted to package it into Nix, but it didn't work on NixOS.

**Comment by eclairevoyant** [comment]
I don't think it's necessary to close this one out, I'm sure it can be patched to work on NixOS.

**Pull request closed without merging**